packages/@aws-cdk/aws-appmesh/lib/virtual-router.ts

```diff
@@ -82,6 +93,41 @@ abstract class VirtualRouterBase extends cdk.Resource implements IVirtualRouter
 
     return route;
   }
+
+  /**
+   * Grants the given entity all read permissions for this VirtualRouter.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeVirtualRouter',
+      'appmesh:ListVirtualRouter',
+      'appmesh:ListTagsForResource',
+    );
+  }
+
+  /**
+   * Grants the given entity all write permissions for this VirtualRouter.
+   */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateVirtualRouter',
+      'appmesh:UpdateVirtualRouter',
+      'appmesh:DeleteVirtualRouter',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
```

Comment:
Same questions here as for `GatewayRoute` - what would necessitate a resource having these permissions?